### DISCUSSION TITLE
Question about capabilities of Casadi for OpenModelica Model import and OCP

### DISCUSSION THREAD

**Discussion by LevinMatz** [casadi/casadi#3300] [Q&A]
Hello,

I am working with non-linear energy system models in OpenModelica and investigating tools for optimal control of such models.

I was thinking that it might be possible in casadi to do the following steps:

1. Import OpenModelica model
2. Disretization of the DAE from the import
3. Optimization of the discetizised NLP

As far as I know, legacy import is not working anymore (documentation 7.2 https://web.casadi.org/docs/#symbolic-import-of-ocps-from-modelica).

Is there a way to do step 1 anyways? I tried to import as an DAE builder object as described in the example script "modelica_fmu_import" but I am not sure how to continue with the resulting object.

Thanks in advance!

**Comment by jaeandersson** [comment]
Symbolic input might still work - it just hasn't been actively maintained and properly tested for some time. And it never worked with very complex structures - sure to limitations on both the OpenModelica side and on the CasADi side.

That being said, an alternative approach is to export standard FMUs from OpenModelica and import using the new FMU import support. This support is still under development.
